pennylane/transforms/unitary_to_rot.py

```diff
@@ -70,13 +76,67 @@ def qfunc():
     >>> print(qml.draw(transformed_qnode)())
      0: ──Rot(-1.35, 1.83, -0.606)──┤ ⟨Z⟩
 
+
+    .. UsageDetails::
+
+        This decomposition is not fully differentiable. We **can** differentiate
+        with respect to input QNode parameters when they are not used to
+        explicitly construct a :math:`4 \times 4` unitary matrix being
+        decomposed. So for example, the following will work:
+
+        .. code-block:: python3
+
+            U = scipy.stats.unitary_group.rvs(4)
+
+            def circuit(angles):
+                qml.QubitUnitary(U, wires=["a", "b"])
+                qml.RX(angles[0], wires="a")
+                qml.RY(angles[1], wires="b")
+                qml.CNOT(wires=["b", "a"])
+                return qml.expval(qml.PauliZ(wires="a"))
+
+            dev = qml.device('default.qubit', wires=["a", "b"])
+            transformed_qfunc = qml.transforms.unitary_to_rot(circuit)
+            transformed_qnode = qml.QNode(transformed_qfunc, dev)
+
+        >>> g = qml.grad(transformed_qnode)
+        >>> g([0.2, 0.3])
+        [array(-0.24454992), array(-0.23913254)]
+
+        However, the following example will **not** be differentiable:
+
+        .. code-block:: python3
+
+            def circuit(angles):
+                z = angles[0]
+                x = angles[1]
+
+                Z_mat = np.array([[np.exp(-1j * z / 2), 0.0], [0.0, np.exp(1j * z / 2)]])
+
+                c = np.cos(x / 2)
+                s = np.sin(x / 2) * 1j
+                X_mat = np.array([[c, -s], [-s, c]])
+
+                U = np.kron(Z_mat, X_mat)
+
+                qml.Hadamard(wires="a")
+
+                # U depends on the input parameters
+                qml.QubitUnitary(U, wires=["a", "b"])
+
+                qml.CNOT(wires=["b", "a"])
+                return qml.expval(qml.PauliX(wires="a"))
     """
+
     for op in tape.operations + tape.measurements:
         if isinstance(op, qml.QubitUnitary):
-            # Only act on single-qubit unitary operations
-            if qml.math.shape(op.parameters[0]) != (2, 2):
-                continue
-
-            zyz_decomposition(op.parameters[0], op.wires[0])
+            # Single-qubit unitary operations
+            if qml.math.shape(op.parameters[0]) == (2, 2):
+                zyz_decomposition(op.parameters[0], op.wires[0])
+            # Two-qubit unitary operations
+            elif qml.math.shape(op.parameters[0]) == (4, 4):
+                two_qubit_decomposition(op.parameters[0], op.wires)
+            else:
+                qml.apply(op)
```

Comment:
This is definitely covered in tests; the circuits run through the transform consist of more than just `QubitUnitary` operations.